tests/utils.py

```diff
@@ -116,52 +193,41 @@ def gdb_start_silent_cmd(cmd: CommandType, before: CommandType = (), after: Comm
 
 def gdb_start_silent_cmd_last_line(cmd: CommandType, before: CommandType = (),
                                    after: CommandType = (),
-                                   target: Path = DEFAULT_TARGET,
+                                   target: pathlib.Path = DEFAULT_TARGET,
                                    strip_ansi=STRIP_ANSI_DEFAULT) -> str:
     """Execute `gdb_start_silent_cmd()` and return only the last line of its output."""
     return gdb_start_silent_cmd(cmd, before, after, target, strip_ansi).splitlines()[-1]
 
 
 def gdb_test_python_method(meth: str, before: str = "", after: str = "",
-                           target: Path = DEFAULT_TARGET,
+                           target: pathlib.Path = DEFAULT_TARGET,
                            strip_ansi: bool = STRIP_ANSI_DEFAULT) -> str:
     brk = before + ";" if before else ""
     cmd = f"pi {brk}print({meth});{after}"
     return gdb_start_silent_cmd(cmd, target=target, strip_ansi=strip_ansi)
 
 
-def include_for_architectures(valid_architectures: Iterable[str] = CI_VALID_ARCHITECTURES):
-    def wrapper(f):
-        def inner_f(*args, **kwargs):
-            if ARCH in valid_architectures:
-                f(*args, **kwargs)
-            else:
-                sys.stderr.write(f"SKIPPED for {ARCH}  ")
-                sys.stderr.flush()
-        return inner_f
-    return wrapper
-
-
-def exclude_for_architectures(invalid_architectures: Iterable[str] = ()):
-    def wrapper(f):
-        def inner_f(*args, **kwargs):
-            if ARCH not in invalid_architectures:
-                f(*args, **kwargs)
-            else:
-                sys.stderr.write(f"SKIPPED for {ARCH}  ")
-                sys.stderr.flush()
-        return inner_f
-    return wrapper
+def gdb_time_python_method(meth: str, setup: str,
+                           py_before: str = "", py_after: str = "",
+                           before: CommandType = (), after: CommandType = (),
+                           target: pathlib.Path = DEFAULT_TARGET,
+                           strip_ansi: bool = STRIP_ANSI_DEFAULT, number: int = 1000) -> float:
+    brk = py_before + ";" if py_before else ""
+    cmd = f"""pi import timeit;{brk}print(timeit.timeit("{meth}", """\
+          f"""setup="{setup}", number={number}));{py_after}"""
+    lines = gdb_run_cmd(cmd, before=before, after=after,
+                        target=target, strip_ansi=strip_ansi).splitlines()
+    return float(lines[-1])
 
 
-def _target(name: str, extension: str = ".out") -> Path:
+def _target(name: str, extension: str = ".out") -> pathlib.Path:
```

Comment:
We have people import and use this function, so it shouldn't be named with a leading underscore. Maybe `get_target_path` or something?